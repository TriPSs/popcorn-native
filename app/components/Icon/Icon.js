import React from 'react'
import PropTypes from 'prop-types'
import RNIcon from 'react-native-vector-icons/MaterialCommunityIcons'

import colors from 'modules/colors'

export const styles = {

  '01DP': {
    shadowOpacity: 1,
    textShadowRadius: 10,
    textShadowOffset: {
      width: 0,
      height: 0,
    },
  },

}

export const Icon = ({ style, color: colorProp, emphasis: emphasisProp, ...rest }) => {
  let emphasis = ''
  const upperColor = colorProp.toUpperCase()

  if (emphasisProp !== 'high') {
    emphasis = emphasisProp === 'medium'
      ? 80
      : 38
  }

  const color = `${colors.ICON[upperColor]}${emphasis}`

  return (
    <RNIcon
      style={[
        style,
        upperColor.includes('DP')
          ? styles[upperColor]
          : {},
      ]}
      color={color}
      {...rest}>
    </RNIcon>
  )
}

Icon.propTypes = {

  color: PropTypes.oneOf([
    'white',
    'black',
    'primary',

    '01dp',
  ]),

  emphasis: PropTypes.oneOf([
    'high',
    'medium',
    'low',
  ]),

  style: PropTypes.object,
}

Icon.defaultProps = {
  emphasis: 'high',
  color: 'white',
}


export default Icon
