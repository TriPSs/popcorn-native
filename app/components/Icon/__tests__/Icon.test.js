import React from 'react'
import { shallow } from 'enzyme'

import Icon from '../Icon'

describe('(Component) Icon', () => {

  it('should render with defaults', () => {
    const result = shallow(
      <Icon name={'settings'} />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with emphasis medium', () => {
    const result = shallow(
      <Icon name={'settings'} emphasis={'medium'} />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with emphasis low', () => {
    const result = shallow(
      <Icon name={'settings'} emphasis={'low'} />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with a different size', () => {
    const result = shallow(
      <Icon name={'settings'} size={15} />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with a different color', () => {
    const result = shallow(
      <Icon name={'settings'} color={'primary'} />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with custom style', () => {
    const result = shallow(
      <Icon name={'settings'} style={{ padding: 20 }} />,
    )

    expect(result).toMatchSnapshot()
  })

})
