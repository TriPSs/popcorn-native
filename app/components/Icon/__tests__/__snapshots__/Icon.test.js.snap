// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Icon should render with a different color 1`] = `
<Icon
  allowFontScaling={false}
  color="#FECC91"
  name="settings"
  size={24}
  style={
    Array [
      null,
      Object {},
    ]
  }
/>
`;

exports[`(Component) Icon should render with a different size 1`] = `
<Icon
  allowFontScaling={false}
  color="#ffffff"
  name="settings"
  size={15}
  style={
    Array [
      null,
      Object {},
    ]
  }
/>
`;

exports[`(Component) Icon should render with custom style 1`] = `
<Icon
  allowFontScaling={false}
  color="#ffffff"
  name="settings"
  size={24}
  style={
    Array [
      Object {
        "padding": 20,
      },
      Object {},
    ]
  }
/>
`;

exports[`(Component) Icon should render with defaults 1`] = `
<Icon
  allowFontScaling={false}
  color="#ffffff"
  name="settings"
  size={24}
  style={
    Array [
      null,
      Object {},
    ]
  }
/>
`;

exports[`(Component) Icon should render with emphasis low 1`] = `
<Icon
  allowFontScaling={false}
  color="#ffffff38"
  name="settings"
  size={24}
  style={
    Array [
      null,
      Object {},
    ]
  }
/>
`;

exports[`(Component) Icon should render with emphasis medium 1`] = `
<Icon
  allowFontScaling={false}
  color="#ffffff80"
  name="settings"
  size={24}
  style={
    Array [
      null,
      Object {},
    ]
  }
/>
`;
